lib/src/http.rs

```diff
@@ -948,22 +948,28 @@ impl Session {
             .borrow()
             .listeners
             .get(&self.listener_token)
+            .with_context(|| "No listener found for this request")? // very unlikely
             .as_ref()
-            .and_then(|listener| listener.borrow().frontend_from_request(host, uri, method));
+            .borrow()
+            .frontend_from_request(host, uri, method);
 
-        let cluster_id = match cluster_id_res {
-            Some(Route::ClusterId(cluster_id)) => cluster_id,
-            Some(Route::Deny) => {
-                self.set_answer(DefaultAnswerStatus::Answer401, None);
-                bail!("Unauthorized route");
-            }
-            None => {
-                let no_host_error = format!("Host not found: {}", host);
+        let route = match cluster_id_res {
+            Ok(route) => route,
+            Err(e) => {
+                let no_host_error = format!("Host not found: {}: {:#}", host, e);
                 self.set_answer(DefaultAnswerStatus::Answer404, None);
                 bail!(no_host_error);
             }
         };
 
+        let cluster_id = match route {
```

Comment:
To me, this should be merge with the above match like:

```rust
match cluster_id_res {
Ok(Router::ClusterId(cluster_id)) => ...
Ok(Route::Deny) => ..
Err(err) => ..
}
```

Reply:
done